app/components/CrashRecommendationCard.tsx

```diff
@@ -0,0 +1,297 @@
+          recommendation_status_id: null,
+          crash_pk,
+          created_by: user?.email,
+          recommendations_partners: [],
+        },
+  });
+
+  const { mutate } = useMutation(
+    recommendation
+      ? UPDATE_RECOMMENDATION_MUTATION
+      : INSERT_RECOMMENDATION_MUTATION
+  );
+
+  const onSave = useCallback(
+    async (data: RecommendationFormInputs) => {
+      setIsMutating(true);
+      // just use a generic object to make the data structuring easier
+      const payload: Record<string, unknown> = data;
+
+      const [addPartners, deletePartnerPks] = getPartnerChanges(
+        recommendation?.id || null,
+        recommendation?.recommendations_partners || [],
+        data.recommendations_partners || []
+      );
+
+      let variables;
+      if (recommendation) {
+        delete payload.recommendations_partners;
+        variables = {
+          record: payload,
+          id: recommendation?.id,
+          addPartners,
+          deletePartnerPks,
+        };
+      } else {
+        payload.recommendations_partners = { data: addPartners };
+        variables = { record: payload };
+      }
+      await mutate(variables, { skip_updated_by_setter: true });
+      await onSaveCallback();
+      setIsMutating(false);
+      setIsEditing(false);
+    },
+    [recommendation, mutate, onSaveCallback, setIsMutating, setIsEditing]
+  );
+
+  return (
+    <Card>
+      <Card.Header>Fatality Review Board recommendations</Card.Header>
+      <Card.Body>
+        <Form id="recommendationForm" onSubmit={handleSubmit(onSave)}>
+          <Form.Group className="mb-3">
+            <Form.Label className="fw-bold">Recommendation</Form.Label>
+            {isEditing && (
+              <Form.Control
+                {...register("rec_text", {
+                  // coerce empty fields to null
+                  setValueAs: (v) => v?.trim() || null,
+                })}
+                as="textarea"
+                rows={6}
+                autoFocus={true}
+              />
+            )}
+            {!isEditing && <p>{recommendation?.rec_text || ""}</p>}
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label className="fw-bold">Update</Form.Label>
+            {isEditing && (
+              <Form.Control
+                {...register("rec_update", {
+                  // coerce empty fields to null
+                  setValueAs: (v) => v?.trim() || null,
+                })}
+                as="textarea"
+                rows={6}
+              />
+            )}
+            {!isEditing && <p>{recommendation?.rec_update || ""}</p>}
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label className="fw-bold">Partners</Form.Label>
+            {!isEditing && (
+              <p>
+                {recommendation?.recommendations_partners
+                  ?.map(
+                    (x) => x.atd__coordination_partners_lkp?.coord_partner_desc
```

Comment:
im kinda anti single letter variables, even though i guess this should be an exception to that rule bc its just a one liner loop?

Reply:
ditto 🙏